README.md

```diff
@@ -42,7 +42,346 @@ Or install it yourself as:
+```
+
+### Engines
+
+The builtin rendering engine is [ERb](http://en.wikipedia.org/wiki/ERuby).
+
+### Root
+
+Template lookup is performed under the `Lotus::Mailer.configuration.root` directory. You can specify a different path on a per mailer basis:
+
+```ruby
+class MailerWithDifferentRoot
+  include Lotus::Mailer
+
+  root 'path/to/root'
+end
+```
+
+### Configuration
+
+__Lotus::Mailer__ can be configured with a DSL that determines its behavior.
+It supports a few options:
+
+```ruby
+require 'lotus/mailer'
+
+Lotus::Maler.configure do
+  # Set the root path where to search for templates
+  # Argument: String, Pathname, #to_pathname, defaults to the current directory
+  #
+  root '/path/to/root'
+
+  # Set the Ruby namespace where to lookup for mailers
+  # Argument: Class, Module, String, defaults to Object
+  #
+  namespace 'MyApp::Mailer'
+```
+
+All those global configurations can be overwritten at a finer grained level:
+`mailers`. Each `mailer` has its own copy of the global configuration, so
+that changes are inherited from the top to the bottom, but not bubbled up in the
+opposite direction.
+
+```ruby
+require 'lotus/mailer'
+
+Lotus::Mailer.configure do
+  root '/path/to/root'
+end
+
+class Show
+  include Lotus::Mailer
+  root '/another/root'
+end
+
+Lotus::Mailer.configuration.root # => #<Pathname:/path/to/root>
+Show.root                      # => #<Pathname:/another/root>
+```
+
+### Delivery Details
+
+When a Ruby object includes __Lotus::Mailer__, a developer can customize delivery details, such as `from`, `to` and `subject`.
+
+The `from` field determines the sender of the email. It can accept a string:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  from "noreply@example.com"
+end
+```
+
+The `from` field can also accept a `Proc` to determine its value:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  from -> { customized_sender }
+
+  def customized_sender
+    "user-#{ user.id }@example.com"
+  end
+end
+```
+
+The `to` field represents one or more emails that will be the recipients of the mail.
+
+It can accept a string:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  to "noreply@example.com"
+end
+```
+
+The to field also accept an array of strings:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  to "[noreply1@example.com, noreply2@example.com]"
+end
+```
+
+`To` field can also be configured using a `Proc` to determine its value:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  to -> { customized_recipient }
+
+  def customized_recipient
+    "user-#{ user.id }@example.com"
+  end
+end
+```
+
+The subject of the message is determined by the field `subject`. It can accept a string:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  subject "This is the subject"
+end
+```
+
+The `subject` field can also accept a `Proc` to determine its value:
+
+```ruby
+class InvoiceMailer
+  include Lotus::Mailer
+
+  subject -> { customized_subject }
```

Comment:
I don't like this syntax. Especially for _to_.
TO/CC/BCC will be not static (hence not via DSL configuration) in the most cases. Subject might be customized per recipient as well.

So why don't we just provide a 

```ruby
def to / bcc / cc
end

def subject
end

def from
end
``` way to configure/override the message instead of requiring a Proc and a method?